iterators.tex

```diff
@@ -4287,6 +4394,173 @@
+\tcode{last}. If \tcode{S} is a literal type, this constructor
+shall be a \tcode{constexpr} constructor.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{move_iterator}!constructor}%
+\begin{itemdecl}
+explicit move_sentinel(S s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Constructs a \tcode{move_sentinel}, initializing
+\tcode{last} with \tcode{s}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{move_sentinel}!constructor}%
+\begin{itemdecl}
+move_sentinel(const move_sentinel<ConvertibleTo<S>>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Constructs a \tcode{move_sentinel}, initializing
+\tcode{last} with \tcode{s.last}.
+\end{itemdescr}
+
+\rSec4[move.sent.op=]{\tcode{move_sentinel::operator=}}
+
+\indexlibrary{\idxcode{operator=}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator=}}%
+\begin{itemdecl}
+move_sentinel& operator=(const move_sentinel<ConvertibleTo<S>>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Assigns \tcode{s.last} to \tcode{last}.
+\end{itemdescr}
+
+\rSec4[move.sent.op.comp]{\tcode{move_sentinel} comparisons}
+
+\indexlibrary{\idxcode{operator==}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator==}}%
+\begin{itemdecl}
+template <class I, Sentinel<I> S>
+  bool operator==(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+template <class I, Sentinel<I> S>
+  bool operator==(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{i.current == s.last}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{operator"!=}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator"!=}}%
+\begin{itemdecl}
+template <class I, Sentinel<I> S>
+  bool operator!=(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+template <class I, Sentinel<I> S>
+  bool operator!=(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{!(i == s)}.
+\end{itemdescr}
+
+\rSec4[move.sent.nonmember]{\tcode{move_sentinel} non-member functions}
+
+\indexlibrary{\idxcode{operator-}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator-}}%
+\begin{itemdecl}
+template <class I, SizedSentinel<I> S>
+  difference_type_t<I> operator-(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{s.last - i.current}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template <class I, SizedSentinel<I> S>
+  difference_type_t<I> operator-(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{i.current - s.last}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{make_move_sentinel}}%
+\begin{itemdecl}
+template <Semiregular S>
+  move_sentinel<S> make_move_sentinel(S s);
```

Comment:
This is a potential pessimization. There should be an overload of `make_move_sentinel` that takes an `InputIterator` and returns a `move_iterator`. That way, a bounded `[I,I)` range is not turned into a non-bounded `[move_iterator<I>,move_sentinel<I>)` range in generic contexts.